frame/oracle/src/lib.rs

```diff
@@ -612,15 +615,15 @@ pub mod pallet {
 			price: T::PriceValue,
 			asset_id: T::AssetId,
 		) {
-			let asset_info = Self::asset_info(asset_id);
+			let asset_info = Self::asset_info(asset_id).unwrap_or_default();
```

Comment:
I think it would be nice to make this consistent with `.ok_or(Error::<T>::InvalidAssetId)?`, such as tweaking the function `Self::asset_info` to directly return a result and never rely on the default implementation, which is basically a null struct?

Reply:
concerned about panics and state changes I thought the best way to handle this would be to pass it down from the initial iteration, also should save state reads too so that is just better

Reply:
:+1: